spirv_hlsl.hpp

```diff
@@ -186,6 +190,9 @@ class CompilerHLSL : public CompilerGLSL
 	void add_hlsl_resource_binding(const HLSLResourceBinding &resource);
 	bool is_hlsl_resource_binding_used(spv::ExecutionModel model, uint32_t set, uint32_t binding) const;
 
+	// Controls which storage buffer bindings will be forced to be declared as UAVs.
+	void set_hlsl_force_storage_buffer_as_uav(ID binding);
```

Comment:
Using ID here is a bit surprising, since everyone always wanted it to be based on set/binding instead. To get IDs you first need to perform reflection, where with set/binding you don't since you emit it straight from a descriptor set layout abstraction. I'd prefer if this API was based around set/binding.

Reply:
Makes sense, SGTM.